fontir/src/variations.rs

```diff
@@ -243,6 +286,41 @@ impl VariationModel {
 
         Ok(result)
     }
+
+    /// Convert relative deltas to absolute values at the given location.
+    ///
+    /// Rust version of <https://github.com/fonttools/fonttools/blob/4ad6b0db/Lib/fontTools/varLib/models.py#L514-L545>
+    pub fn interpolate_from_deltas<V>(
+        &self,
+        location: &NormalizedLocation,
+        deltasets: &[(VariationRegion, Vec<V>)],
+    ) -> Vec<V>
+    where
+        V: Copy + Mul<f64, Output = V> + Add<V, Output = V>,
+    {
+        let mut result = None;
+        for (region, deltas) in deltasets.iter() {
+            let scalar = region.scalar_at(location).into_inner() as f64;
+            if scalar == 0.0 {
+                continue;
+            }
+            let contribution = deltas.iter().map(|d| *d * scalar).collect::<Vec<_>>();
+            if result.is_none() {
+                result = Some(contribution);
+            } else {
+                result = Some(
+                    result
+                        .unwrap()
+                        .iter()
+                        .zip(contribution.into_iter())
+                        .map(|(r, c)| *r + c)
+                        .collect(),
```

Comment:
this is allocating a new vec to store the result on each iteration, but it would be more efficient to just modify the existing values in place.